### PR TITLE
Replace Algolia Docsearch with Inkeep

### PR DESCRIPTION
**Pull request by ysadamt**
## Summary

- Remove Algolia Docsearch component and its dependencies `@docsearch/js` and `@docsearch/css`
- Add Inkeep component and dependencies
- Replace fonts and change search bar style to match previous component
- Fixes GRO-132

## How did you test this change?

|Before|After|
|--|--|
|image|image|
|image|imageimage|

## Are there any deployment considerations?

No

## Does this work require review from our design team?

No

### CODE REVIEW

File: highlight.io/utils/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+type InkeepSharedSettings = {
+	baseSettings: InkeepBaseSettings
+	aiChatSettings: InkeepAIChatSettings
+	searchSettings: InkeepSearchSettings
+	modalSettings: InkeepModalSettings
+}
+
+const useInkeepSettings = (): InkeepSharedSettings => {
+	const baseSettings: InkeepBaseSettings = {
+		apiKey: '3721b3f00a2e161d2eb143932695e984a9db05ffb89bdb57',
```
[the remark this answers is not in the document]
Comment (Vadman97) [reply highlight.io/utils/hooks.ts:17]: client side public key

**Comment by Vadman97** [comment]
hey @ysadamt, can you rebase on `origin/main`? not sure if these failures are related, but perhaps caused by some dep upgrade? https://github.com/highlight/highlight/actions/runs/10727178199/job/29748894718?pr=9261

**Review by Vadman97 — APPROVED** [review]
lgtm 🎉 pending the build failures

**Comment by ysadamt** [comment]
> hey @ysadamt, can you rebase on `origin/main`? not sure if these failures are related, but perhaps caused by some dep upgrade? https://github.com/highlight/highlight/actions/runs/10727178199/job/29748894718?pr=9261

I seem to be getting the same typegen errors in `yarn test:all` locally even after rebasing to `origin/main`, are these expected?

**Comment by Vadman97** [comment]
@mayberryzane any idea why the build might be failing? is it failing on main?

**Comment by ysadamt** [comment]
I seem to getting this error as well whenever the page loads, not sure if it's related: 
image

**Comment by ysadamt** [comment]
bumping this for build failures @mayberryzane @Vadman97

**Comment by Vadman97** [comment]
looks like this is the cuplit, @apollo/client had a breaking API change in between 3.7.0 and 3.11.8. fixing now
Screenshot 2024-09-13 at 15 29 34

**Pull request merged**